### ISSUE TITLE
Modify gradient descent for specific layer

### ISSUE THREAD

**Issue by briliantnugraha** [keras-team/keras#9584]
Dear Keras-team and @fchollet, thanks for your great contribution so that the implementation of NN becomes much easier to produce.

I have a question, how do we modify the gradient in Keras's backward propagation pass for specific layer?

I have checked the solution in Tensorflow,
It could be achieved by calling all of the tensor variables manually through:
. tf.train.Optimizer.compute_gradients, then
. tf.train.Optimizer.apply_gradients
the solution that I have seen so far:
url 1: https://stackoverflow.com/questions/46125848/modify-gradient-in-tensorflow-backward-pass
url 2: https://stackoverflow.com/questions/43839431/tensorflow-how-to-replace-or-modify-gradient/43948872

example of model's structure and mark of edit layer:
model = Sequential()
model.add(Conv2D(32, kernel_size=(3, 3),
                 activation='relu',
                 input_shape=input_shape))
model.add(Conv2D(64, (3, 3), activation='relu')) #want to edit this layer
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))
model.add(Flatten())
model.add(Dense(128, activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(num_classes, activation='softmax'))

model.compile(loss=keras.losses.categorical_crossentropy,
              optimizer=keras.optimizers.Adadelta(),
              metrics=['accuracy'])
as in keras example on https://github.com/keras-team/keras/blob/master/examples/mnist_cnn.py

Is there a way to link this method to Keras's back-prop step? Or is there a simpler way to do it?
Big thanks for the help, cheers~
